packages/fastbtc-node/src/core/initiator.ts

```diff
@@ -0,0 +1,211 @@
+import {MessageUnion, Network, Node} from 'ataraxia';
+import {sleep} from '../utils';
+import NodeValues from '../utils/nodevalues';
+import Logger from '../logger';
+
+interface SyncInitiatorMessage {
+    initiatorId: string | null;
+}
+interface InitiatorVotingMessage {
+    'fastbtc:initiator:sync-request': SyncInitiatorMessage,
+    'fastbtc:initiator:sync-response': SyncInitiatorMessage,
+}
+
+/**
+ * A class for deciding the initiator of the network.
+ *
+ * The first version only contains support for a "sticky" initiator so it doesn't change randomly.
+ * Further versions should also support rotating the initiator.
+ */
+export class InitiatorVoting {
+    private logger = new Logger('initiatorvoting');
+    private nodeInitiatorIds = new NodeValues();
+    private updateIntervalId?: NodeJS.Timer;
+
+    constructor(
+        private network: Network<InitiatorVotingMessage>,
+    ) {
+        network.onNodeUnavailable(this.onNodeUnavailable)
+        network.onMessage(this.onMessage);
+    }
+
+    /**
+     * Start updating state from the network periodically
+     */
+    public async start() {
+        if (this.updateIntervalId) {
+            this.logger.warning('Already started, not starting again');
+            return;
+        }
+        await this.syncInitiatorFromNetwork();
+        this.updateIntervalId = setInterval(() => {
```

Comment:
instead of using setInterval you really should use a while loop - what happens if there is a > 10 second delay in something